src/components/SettingsListItem.js

```diff
@@ -0,0 +1,127 @@
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+import { ActivityIndicator, StyleSheet, View } from 'react-native';
+import { ListItem } from 'react-native-elements';
+
+import { colors, normalize } from '../config';
+import { Icon } from './Icon';
+import { Switch } from './Switch';
+import { BoldText, RegularText } from './Text';
+import { Touchable } from './Touchable';
+import { WrapperHorizontal, WrapperRow } from './Wrapper';
+
+const previewListItem = {
+  title: 'Proident tempor aliqua',
+  name: 'Id duis nisi reprehenderit ut',
+  category: 'Lorem ipsum',
+  image: 'https://via.placeholder.com/400.png/bcbbc1/fff?text=Lorem+ipsum',
+  topDivider: true
+};
+
+// TODO: toggle mit alert für matomo
+// TODO: snack bar / toast als nutzerinfo
+// TODO: overlay for selecting text list type: https://reactnativeelements.com/docs/1.2.0/overlay
+export const SettingsListItem = ({ item, index, section, orientation, dimensions }) => {
+  const {
+    title,
+    bottomDivider,
+    topDivider,
+    type,
+    listSelection,
+    Component,
+    value,
+    onActivate,
+    onDeactivate
+  } = item;
+
+  const [loading, setLoading] = useState(false);
+  const [switchValue, setSwitchValue] = useState(value);
+
+  const toggleSwitch = async (newSwitchValue) => {
+    setLoading(true);
+
+    setSwitchValue(newSwitchValue);
+
+    if (newSwitchValue && onActivate) {
+      await onActivate(() => setSwitchValue(false));
+    } else if (!newSwitchValue && onDeactivate) {
+      await onDeactivate(() => setSwitchValue(true));
```

Comment:
I am slightly confused as to why the `onActivate` and `onDeactivate` should take a function as an argument.

it seems like 
```ts
await onActivate();
setSwitchValue(false);
```
would be a better approach.

Is there a specific benefit that I am missing?

Reply:
yes, the method is the `revert` method called when declining in the alert, as the main state change happens before.
so we should consider changing that, to avoid the need of a revert?